src/app/space/create/codebases/codebases-item-actions/codebases-item-actions.component.ts

```diff
@@ -51,28 +50,41 @@ export class CodebasesItemActionsComponent implements OnDestroy, OnInit {
    */
   createAndOpenWorkspace(): void {
     this.workspaceBusy = true;
-    this.subscriptions.push(this.workspacesService.createWorkspace(this.codebase.id)
-      .subscribe(workspaceLinks => {
-        this.workspaceBusy = false;
-        if (workspaceLinks != undefined) {
-          let name = this.getWorkspaceName(workspaceLinks.links.open);
-          this.windowService.open(workspaceLinks.links.open, name);
-
-          this.notifications.message({
-            message: `Workspace created!`,
-            type: NotificationType.SUCCESS
-          } as Notification);
-
-          // Poll for new workspaces
-          this.broadcaster.broadcast('workspaceCreated', {
-            codebase: this.codebase,
-            workspaceName: name
+    this.subscriptions.push(this.cheService.getState().switchMap(che => {
+      if (!che.clusterFull) {
+        // create
+        return this.workspacesService
+          .createWorkspace(this.codebase.id)
+          .map(workspaceLinks => {
+            this.workspaceBusy = false;
+            if (workspaceLinks != undefined) {
+              let name = this.getWorkspaceName(workspaceLinks.links.open);
+              this.windowService.open(workspaceLinks.links.open, name);
+              this.notifications.message({
+                message: `Workspace created!`,
+                type: NotificationType.SUCCESS
+              } as Notification);
+              // Poll for new workspaces
+              this.broadcaster.broadcast('workspaceCreated', {
+                codebase: this.codebase,
+                workspaceName: name
+              });
+            } else {
+              // display error message
+              this.notifications.message({
+                message: `Workspace error during creation.`,
+                type: NotificationType.DANGER
+              } as Notification);
+            }
           });
-        }
-      }, error => {
-        this.workspaceBusy = false;
-        this.handleError('Failed to create workspace', NotificationType.DANGER);
-      }));
+      } else {
+        // display error message
+        this.notifications.message({
+          message: `Workspace capacity is reached, new workspace cannot be started.`,
```

Comment:
@corinnekrych could you please change the message to `OpenShift Online cluster is currently out of capacity, workspace cannot be started`  ?